### DISCUSSION TITLE
graphql/codegen/plugins/typescript/typescript-msw

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9972] [Docs Discussions]
# graphql/codegen/plugins/typescript/typescript-msw

https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-msw

**Comment by charlieforward9** [comment]
Is this up to date?

**Reply by charlieforward9** [reply]
This codegen does not automatically import the dependencies it uses. How do I configure this to handle automatically?

**Reply by ikibalnyi** [reply]
I managed to solve it using the [import-types-preset](https://the-guild.dev/graphql/codegen/plugins/presets/import-types-preset).

```
pnpm add -D @graphql-codegen/import-types-preset
```

and then:
```js
// codegen.js
    'src/graphql/generated.ts': {
      // regular configuration of your codegen client types
    },
    'src/graphql/mocks.ts': {
      schema: schema,
      preset: 'import-types',
      presetConfig: {
        typesPath: './generated',
      },
      documents: ['src/**/*.graphql'],
      plugins: ['typescript-msw'],
    },
```

**Reply by charlieforward9** [reply]
This solved my problem and actually made my import statement a lot more concise since it used the `import * as ...` instead of importing all the individual types. Thank you!

**Comment by tlmader** [comment]
Has anyone created a setup where we get generated msw handlers + default mock data (like in the typescript-mock-data plugin)? Would be a convenient setup if you only needed to return the fields that matter for the test.
